beacon-chain/blockchain/execution_engine.go

```diff
@@ -213,6 +213,9 @@ func (s *Service) notifyNewPayload(ctx context.Context, postStateVersion int,
 			"invalidCount": len(invalidRoots),
 		}).Warn("Pruned invalid blocks")
 		return false, invalidBlock{ErrInvalidPayload}
+	case powchain.ErrInvalidBlockHashPayloadStatus:
```

Comment:
This isn't fully correct, I think you also need to prune and delete it the same way for `INVALID` blocks. cc @potuz

Reply:
Missed that. Adding it now